pkg/volume/util/subpath/subpath_linux.go

```diff
@@ -160,7 +160,7 @@ func getSubpathBindTarget(subpath Subpath) string {
 	return filepath.Join(subpath.PodDir, containerSubPathDirectoryName, subpath.VolumeName, subpath.ContainerName, strconv.Itoa(subpath.VolumeMountIndex))
 }
 
-func doBindSubPath(mounter mount.Interface, subpath Subpath) (hostPath string, err error) {
+func doBindSubPath(mounter mount.Interface, subpath Subpath, mountWithMounterInstance bool) (hostPath string, err error) {
```

Comment:
I added this flag because of the unit test, the unit test uses a fake `mounter` however the implementation uses the logic inside `subpath_mount_linux.go` which doesn't need a mounter. Luckily this method is private so the places where it's used are scoped to this package.

Another approach if we want to use the interface that can't be modified in k8s.io/utils/mount is to copy k8s.io/utils/mount to the kubernetes codebase similar to what's done in master through `/staging`

Reply:
how hard is it to actually add a new Mounter implementation with only the necessary methods implemented?

Reply:
This is where the mounter instance is created

https://github.com/kubernetes/kubernetes/blob/9d49ff96ca7491b8f2736aa1be73a2ed66ef08cc/cmd/kubelet/app/server.go#L365-L366

I could create an instance that's only used by subpath here that implements the necessary methods as you say, other places like

https://github.com/kubernetes/kubernetes/blob/9d49ff96ca7491b8f2736aa1be73a2ed66ef08cc/cmd/kubelet/app/server.go#L393

 would keep on using the k8s.io/utils/mount instance, so the change would be

```
	mounter := mount.New(s.ExperimentalMounterPath)
    subpathMounter := subpathmount.New(s.ExperimentalMounterPath)
	subpather := subpath.New(subpathMounter)

    // ...

    return &kubelet.Dependencies{
		Mounter:             mounter,
		Subpather:           subpather
    }
```

Reply:
I think that would be a cleaner approach.

cc @jsafrane any opinions?

Reply:
Are you suggesting some `MounterWithMountArgs` implementation of mounter? That would work.

Reply:
I implemented this approach in https://github.com/kubernetes/kubernetes/pull/104340 PTAL, I'm keeping this PR open on the side just in case